tools/run_tests/performance/scenario_config.py

```diff
@@ -145,14 +140,15 @@ def _ping_pong_scenario(name, rpc_type,
   if use_generic_payload:
     if server_type != 'ASYNC_GENERIC_SERVER':
       raise Exception('Use ASYNC_GENERIC_SERVER for generic payload.')
-    scenario['client_config']['payload_config'] = EMPTY_GENERIC_PAYLOAD
-    scenario['server_config']['payload_config'] = EMPTY_GENERIC_PAYLOAD
-  else:
-    # For proto payload, only the client should get the config.
-    scenario['client_config']['payload_config'] = EMPTY_PROTO_PAYLOAD
+  scenario['client_config']['payload_config'] = _payload_type(use_generic_payload, req_size, resp_size)
+  scenario['server_config']['payload_config'] = _payload_type(use_generic_payload, req_size, resp_size)
```

Comment:
I think the problem here is that payload_config should only be set for client for non-generic server. The reason is that the non-generic server (Async and Sync) are supposed to respond with a message size that corresponds to what is requested in SimpleRequest (which is constructed based on client's payload config).

Setting payload config for a server should be illegal (and so C# qps worker is asserting that it is only set for generic server), which is what's breaking C# benchmarks (https://github.com/grpc/grpc/issues/9587).

https://github.com/grpc/grpc/blob/master/src/proto/grpc/testing/messages.proto#L76

https://github.com/grpc/grpc/blob/2e096b0fec7d56a71dc0bcb4ca5c9b6b918a330a/src/proto/grpc/testing/control.proto#L143

https://github.com/grpc/grpc/blob/f9b87301aada7bd80f8487bb0527bfb4c0eaa034/test/cpp/qps/server_async.cc#L366